tinygrad/codegen/linearizer.py

```diff
@@ -50,9 +55,11 @@ def const(self, b:Union[int,float], dtype=dtypes.int32, insert_before=None) -> U
 
   def cast(self, val: UOp, dtype) -> UOp: return self.uop(UOps.CAST, dtype, (val,)) if val.dtype != dtype else val
 
-  def get_reduce_acc(self, op, dtype:DType):
-    if op == ReduceOps.SUM: return 0.0 if dtypes.is_float(dtype) else 0
-    elif op == ReduceOps.MAX: return -math.inf if dtypes.is_float(dtype) else -2**31 if dtypes.is_int(dtype) else False
+  def get_reduce_acc(self):
+    assert self.reduceop
```

Comment:
why this assert?

Reply:
mypy, reduceop is nullable
i could also cast tho

Reply:
you can still pass the self.reduceop here tho and it should be clear to mypy that self.reduceop is not None is that case? otherwise why was this not an issue before?

Reply:
yeah that's becuase then the arg was untyped

passing self.reduceop is repetitive when we know there's only one reduceop in a kernel

Reply:
yeah that's fine for this pr but these stateful helper functions need to be clean up. can you add an assertion message `reduceop cannot be None` and check for `is None`?